hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueue.kt

```diff
@@ -54,8 +62,100 @@ class HmppsQueue(
 
   private fun getRedrivePolicyAttribute(url: String) = GetQueueAttributesRequest.builder().queueUrl(url).attributeNames(QueueAttributeName.REDRIVE_POLICY).build()
 
-  private companion object {
-    val log = LoggerFactory.getLogger(this::class.java)
+  companion object {
+    internal val log: Logger = LoggerFactory.getLogger(this::class.java)
     private val gson = GsonBuilder().setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE).create()
   }
 }
+
+/**
+ * Publishes an event to the specified SQS queue with retry and back off policies.
+ *
+ * Examples:
+ *
+ * Scenario where API is called from UI, and we want a reasonable number of retries while the user waits for a response:
+ * We can just use the default of 4 attempts that would block the thread for around 7 seconds
+ * ```
+ * hmppsQueue.sendMessage(
+ *         eventType = event.eventType,
+ *         event = event.body,
+ *       )
+ * ```
+ *
+ * Scenario where retrying is already done by the calling code; for instance an SQS Listener that transforms events that is idempotent so the entire process can be rerun:
+ * Given we don't want to tie up the listener thread there is no advantage of retrying on this thread so
+ *
+ * ```
+ * hmppsQueue.sendMessage(
+ *       eventType = event.eventType,
+ *       event = event.body,
+ *       retryPolicy = NeverRetryPolicy(),
+ *     )
+ * ```
+ *
+ *
+ * Scenario where thread can hang for longer with more retries
```

Comment:
I really like the real life examples from above - could we add real life examples to this?